servers/publikator/lib/github/index.js

```diff
@@ -132,22 +145,156 @@ module.exports = {
+            owner: $login
+            name: $repoName
+          ) {
+            object(oid: $sha) {
+              ... on Commit {
+                oid
+                author {
+                  email
+                  name
+                }
+                parents (first: 100){
+                  nodes {
+                    oid
+                  }
+                }
+                message
+                committedDate
+              }
+            }
+          }
+        }
+      `,
+      variables: {
+        login,
+        repoName,
+        sha
+      }
     })
-      .then(response => response ? response.data : response)
-      .then(commit => commitNormalizer({
-        ...commit,
-        repo
-      }))
-      .then(async (commit) => {
-        await redis.setAsync(redisKey, JSON.stringify(commit))
-        return commit
-      })
+
+    if (!rawCommit) {
+      return null
+    }
+    const commit = commitFromGQL(repo, rawCommit)
+    await redis.setAsync(redisKey, JSON.stringify(commit))
+    return commit
+  },
+  getCommits: async (repo, { first = 15, after, before }) => {
+    const { githubApolloFetch } = await createGithubClients()
+    const [login, repoName] = repo.id.split('/')
+    const {
+      data: {
+        repository: {
+          refs: {
+            nodes: heads
+          }
+        }
+      }
+    } = await githubApolloFetch({
+      query: `
+        query repository(
+          $login: String!,
+          $repoName: String!,
+          $maxRefs: Int
+          $maxCommits: Int,
+          $commitsSince: GitTimestamp,
+          $commitsUntil: GitTimestamp
+        ) {
+          repository(
+            owner: $login
+            name: $repoName
+          ) {
+            refs(refPrefix: "refs/heads/", first: $maxRefs) {
+              nodes {
+                name
+                target {
+                  ... on Commit {
+                    oid
+                    author {
+                      date
+                    },
+                    history(first: $maxCommits, until: $commitsUntil, since: $commitsSince) {
+                      pageInfo {
+                        hasNextPage
+                      }
+                      totalCount
+                      nodes {
+                        ... on Commit {
+                          oid
+                          author {
+                            email
+                            name
+                          }
+                          parents (first: 100){
+                            nodes {
+                              oid
+                            }
+                          }
+                          message
+                          committedDate
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      `,
+      variables: {
+        login,
+        repoName,
+        maxRefs: 100,
```

Comment:
what will happen once magazine front hit 100 branches? (currently at 95)

Reply:
This is actually just ported from the previous implementation. Can we support more than 100 branches? Can we do it in the UI?